omnifit/spectrum/spectrum.py

```diff
@@ -4,765 +4,840 @@
+        newY = self.y[iSub]
+        self.x = newX
+        self.y = newY
+
+    @clonable
+    def interpolate(self, target_spectrum):
+        """
+        interpolate(target_spectrum,clone=False)
+
+        Interpolate spectrum to match target spectrum resolution.
+        Does not modify current spectrum, but returns a new one, which is
+        a copy of the current spectrum but with the interpolated data on
+        the x and y fields.
+        The target spectrum has to be using the same units on the x and
+        y axes as the current spectrum, or the interpolation will fail.
+
+        Parameters
+        ----------
+        target_spectrum : `BaseSpectrum`
+            The target spectrum which the x axis resolution of the current
+            spectrum should be made to match.
+        clone : `bool`, optional
+            If set to True, returns a modified copy of the spectrum instead
+            of operating on the existing spectrum.
+        """
+        if self.x.unit != target_spectrum.x.unit:
+            raise u.UnitsError('Spectrums have different units on x axis!')
+        if self.y.unit != target_spectrum.y.unit:
+            raise u.UnitsError('Spectrums have different units on y axis!')
+        newX = target_spectrum.x
+        newY = np.interp(newX, self.x, self.y)*self.y.unit
+        self.x = newX
+        self.y = newY
+        self.name = '{0}(interpolated: {1})'.format(
+            self.name,
+            target_spectrum.name)
+
+    def yat(self, x):
+        """
+        yat(x)
+
+        Return interpolated value of y at requested x.
+
+        Parameters
+        ----------
+        x : `float`
+            The x axis coordinate of interest.
+
+        Returns
+        -------
+        The interpolated value of y at the requested x coordinate.
+
+        """
+        return np.interp(x, self.x.value, self.y.value)
+
+    @clonable
+    def convolve(self, kernel, **kwargs):
+        """
+        convolve(kernel, clone=False, **kwargs)
+
+        Use `astropy.convolution.convolve` to convolve the y axis data of the
+        spectrum with the given kernel.
+
+        Parameters
+        ----------
+        kernel : `numpy.ndarray` or `astropy.convolution.Kernel`
+            A convolution kernel to feed into the convolution function.
+        clone : `bool`, optional
+            If set to True, returns a modified copy of the spectrum instead
+            of operating on the existing spectrum.
+        **kwargs : Arguments, optional
+            This can be used to pass additional arguments
+            to `astropy.convolution.convolve`.
+        """
+        if self.convolved:
+            warnings.warn(
+                'Spectrum '+self.name+' has already been convolved once!',
+                RuntimeWarning)
+        yunit = self.y.unit  # to preserve units through convolution
+        self.y = convolution.convolve(self.y, kernel, **kwargs)
+        self.y = self.y * yunit  # restore y units
```

Comment:
```suggestion
        self.y = convolution.convolve(self.y, kernel, **kwargs)
```
Same as above, just let `u.Quantity` handle everything. :)

Reply:
Fixes implemented in 275c50f ; thank you :)